### DISCUSSION TITLE
Support for NixOS stable (23.11)

### DISCUSSION THREAD

**Discussion by crabdancing** [Jovian-Experiments/Jovian-NixOS#343] [General]
As it currently stands, using Jovian-NixOS for a system generally means I have to use whatever unstable version is pinned. This is a problem because the unstable version of NixOS is... unstable. The ability to use Jovian on the stable version would be great!

**Comment by K900** [comment]
Unstable isn't really unstable as in buggy, it's unstable as in may contain breaking changes. The reason we don't support 23.11 is that pretty much everyone is running unstable, and you can see from the git history that regressions are very rare.

**Comment by crabdancing** [comment]
My experience is that my Jovian Steamdeck config has a lot more bugs and glitches than my other NixOS systems. E.g., KDE stuff seems to be breaking more.

**Comment by K900** [comment]
Then please file issues for that.

**Comment by crabdancing** [comment]
Filing issues will only MAYBE fix those specific issues. Otherwise there would be no distinction between stable and unstable, no?

Anyways, the reason I haven't is because they're probabilistic bugs that only crop up under specific circumstances, and I literally do not know what to tell them. But in the meantime, it would be nice is my work devices didn't have to use unstable. :P

**Comment by crabdancing** [comment]
Yeah, like today I found out it was broken:

```nix
building the system configuration...
error:
       … while calling the 'seq' builtin

         at /nix/store/8zgn73r04imfsn9sjynvp0nkkwppxngf-source/lib/modules.nix:322:18:

          321|         options = checked options;
          322|         config = checked (removeAttrs config [ "_module" ]);
             |                  ^
          323|         _module = checked (config._module);

       … while calling the 'throw' builtin

         at /nix/store/8zgn73r04imfsn9sjynvp0nkkwppxngf-source/lib/modules.nix:298:18:

          297|                     ''
          298|             else throw baseMsg
             |                  ^
          299|         else null;

       error: The option `services.displayManager' does not exist. Definition values:
       - In `/nix/store/ixc48z15n22d4kjrgx858pfbdsbs6ap6-source/modules/steam/autostart.nix':
           {
             _type = "if";
             condition = true;
             content = {
               _type = "if";
           ...
 ```
 
This the sort of thing that happens all the time in my experience.

**Comment by K900** [comment]
Yes, because again, we only support and test unstable. I've converted this issue into a discussion, because I don't think we have the bandwidth to support two branches of NixOS as it stands.

**Comment by radioaddition** [comment]
is there perhaps some workaround so that jovian will use unstable stuff while everything else is on stable?

**Reply by K900** [reply]
No, that's not practical.

**Reply by crabdancing** [reply]
_Kind_ of?

Every six months, in time with the release cycle, there is a magical time at which nixos-unstable effectively _is_ stable. I personally just pin Jovian-NixOS to the the version from that time, and only update Jovian-NixOS on a six month cycle. It sucks, but whatever. :P

**Reply by K900** [reply]
_Please_ do not do that. There's been security fixes since that time.

**Reply by crabdancing** [reply]
> _Please_ do not do that. There's been security fixes since that time.

In Jovian-NixOS or in upstream nixpkgs?

**Reply by K900** [reply]
Both.
